### PR TITLE
Feature/em 1331 tagging comments

### PR DESCRIPTION
**Pull request by fazanki**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts [hunk 1]
```diff
@@ -4,12 +4,13 @@ import { AnnotationApiService } from '../annotation-api.service';
 import { AnnotationSet } from './annotation-set.model';
 import { ToolbarEventService } from '../../toolbar/toolbar-event.service';
 import { ViewerEventService } from '../../viewers/viewer-event.service';
-import { Subscription } from 'rxjs';
+import {BehaviorSubject, of, Subscription} from 'rxjs';
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: is this used in this file?

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts [hunk 2]
```diff
@@ -4,12 +4,13 @@ import { AnnotationApiService } from '../annotation-api.service';
...
 import { PageEvent } from '../../viewers/pdf-viewer/pdf-js/pdf-js-wrapper';
 import { AnnotationEventService, SelectionAnnotation } from '../annotation-event.service';
 import { CommentService } from '../comment-set/comment/comment.service';
 import { TextHighlightCreateService } from './annotation-create/text-highlight-create.service';
 import { BoxHighlightCreateService } from './annotation-create/box-highlight-create.service';
+import {take} from 'rxjs/operators';
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: is this used?

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts [hunk 3]
```diff
@@ -112,6 +126,9 @@ export class CommentComponent implements OnChanges {
     this.editable = false;
     this.fullComment = this.originalComment;
     this.changes.emit(false);
+    if (!this.author && !this.fullComment) {
+      this.delete.emit(this._comment);
+    }
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts]: why do we delete the comment on a get?

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts [hunk 4]
```diff
@@ -31,26 +36,34 @@ export class CommentComponent implements OnChanges {
...
+    this.autocompleteTagItems = allTags ? allTags : [];
+  }
+  tagItems: TagItemModel[];
   @ViewChild('form') form: ElementRef;
   @ViewChild('textArea') textArea: ElementRef;
 
   constructor(
-    private readonly commentService: CommentService
+    private readonly commentService: CommentService,
+    private http: HttpClient,
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts]: is this used here?

File: projects/media-viewer/src/lib/annotations/models/tag-item.model.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export interface TagItemModel {
+  name: string;
+  label: string;
+  color?: string;
+  di: string;
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/models/tag-item.model.ts]: should this be id or is it an abbreviation?

File: projects/media-viewer/src/lib/annotations/services/tags/tags.services.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+@Injectable()
+export class TagsServices {
+
+  public tagItems: {[id: string]: TagItemModel[]};
+
+  constructor(private http: HttpClient) {}
+
+  public getAllTags(createdBy): Observable<TagItemModel[]> {
+    const url = `/em-anno/tags/${createdBy}`;
+    return this.http.get<TagItemModel[]>(url);
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/services/tags/tags.services.ts]: maybe this should go to the annotations-api service, as that's where we have all the calls to the annotations API backend

File: projects/media-viewer/src/lib/annotations/services/tags/tags.services.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    this.tagItems = {
+      ...this.tagItems,
+      [annoId]: snakeCasedWithId
+    };
+    console.log(this.tagItems);
+  }
+
+  private snakeCase = string => {
+    return string.replace(/\W+/g, " ")
+      .split(/ |\B(?=[A-Z])/)
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/services/tags/tags.services.ts]: not very clear what this is doing. maybe if you put the regex in variables with names that describe what they're for that'll help

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts (hunk 1 above)
Comment (fazanki) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: nope fixed

File: projects/media-viewer/src/lib/annotations/models/tag-item.model.ts (hunk 5 above)
Comment (fazanki) [reply projects/media-viewer/src/lib/annotations/models/tag-item.model.ts]: removed

**Pull request merged**